packages/datadog-instrumentations/src/cucumber.js

```diff
@@ -0,0 +1,82 @@
+'use strict'
+
+const { addHook, channel } = require('./helpers/instrument')
+const shimmer = require('../../datadog-shimmer')
+
+const runStartCh = channel('ci:cucumber:run:start')
+const runEndCh = channel('ci:cucumber:run:end')
+const runAsyncEndCh = channel('ci:cucumber:run:async-end')
+const runStepStartCh = channel('ci:cucumber:run-step:start')
+const runStepEndCh = channel('ci:cucumber:run-step:end')
+const errorCh = channel('ci:cucumber:error')
+
+function wrapRun (pl, isLatestVersion) {
+  shimmer.wrap(pl.prototype, 'run', run => function () {
+    if (!runStartCh.hasSubscribers) {
+      return run.apply(this, arguments)
+    }
+
+    runStartCh.publish({ pickleName: this.pickle.name, pickleUri: this.pickle.uri })
+    try {
+      const promise = run.apply(this, arguments)
+      promise.finally(() => {
+        runAsyncEndCh.publish({ result: this.getWorstStepResult(), isLatestVersion })
```

Comment:
Why does the subscriber need to know this? Any way to avoid the subscriber having to know which version is used? Can the message be normalized at the publisher level?

Reply:
The idea being that new versions of the library should not require changes to the subscriber.

Reply:
that's good feedback. Applied it in https://github.com/DataDog/dd-trace-js/pull/1860/commits/eb53b701d45aa7257ecc04dcc9c68fa8da6f62a1